import "goby/moos/protobuf/goby_moos_app.proto";
import "goby/common/protobuf/option_extensions.proto";

package gps_proto;

message PackagerConfig
{
  optional GobyMOOSAppConfig common = 1 [(goby.field).description="Configuration common to all Goby MOOS applications"]; // see goby_moos_app.proto
  repeated FieldSpecifics array = 2;
}

message FieldSpecifics
{
  required string class_name = 1;
  required string moos_var = 2; // the string with which the field is associated
  optional string src_field = 3; // string representing the name of this class's source field
  required string dest_field = 4; // same, but for destination
}